### ISSUE TITLE
Support custom feature configuration

### ISSUE THREAD

**Issue by gnunicorn** [ligthyear/clippy-service#17]
[Quoting @mcarton on #1](https://github.com/ligthyear/clippy-service/issues/7#issuecomment-192340051):

> Also tested it, it reports warning in a file guarded by
> 
> `#![cfg_attr(feature="clippy", allow(float_cmp))]`
> 
> Clippy is often used with a `clippy`, `dev`, `nightly`, `testing` or whatever feature in the wild, does the service have a way to compile with a given feature?

**Comment by gnunicorn** [comment]
And to quote myself:

> @Manishearth , @mcarton could you elaborate why the `conf_attr` things doesn't work in the rustc approach I am currently taking? And maybe what we can do about that? Is this because the we don't pass that config? I was thinking of allowing some customization of the rustc command via a `.clippy.toml`in the same folder. There the owner could configure they want "feature" xyz, would that solve the issue?

**Comment by mcarton** [comment]
Note that Clippy might get a `Clippy.toml` file by itself (see https://github.com/Manishearth/rust-clippy/pull/698). Adding another `.clippy.toml` file would be confusing at best. If this is the way we choose, we could just add the conf there.

**Comment by gnunicorn** [comment]
@mcarton: good point. Will that config contain a way to configure said features from there?

**Comment by mcarton** [comment]
No, Clippy cannot change the features itself.

**Comment by mrmonday** [comment]
Another option might be to include it in the URL, along with the repo and branch.

**Comment by Manishearth** [comment]
At the same time, we could allow a `[service] clippy-feature="nightly"` key in the toml. Clippy itself can ignore it, but clippy-service reads it. No reason we can't share a conf file :smile:

**Comment by mcarton** [comment]
> No reason we can't share a conf file

Of course, but with the current PR any unknown key is an error. Should I just white-list `[service]` (but then what about other tools?) or ignore all unknown keys (which means eg. spelling mistake in the conf file will be hard to notice).

**Comment by Manishearth** [comment]
Nah, just whitelist `[service]`. Or we'll call it `[third-party]` or `[external]`

**Comment by mcarton** [comment]
Ok then, I’ve white-listed `third-party`, I preferred that name.

**Comment by gnunicorn** [comment]
Well, I wait for https://github.com/Manishearth/rust-clippy/pull/698 to be merged then. After that I will tackle this as part of the [config-file milestone](https://github.com/ligthyear/clippy-service/milestones/Config%20File)!

**Comment by indiv0** [comment]
https://github.com/Manishearth/rust-clippy/pull/698 has been merged. Any update on this?

**Comment by gnunicorn** [comment]
@indiv0 no work on this yet (from my side), but I am accepting Pull-Requests :-) !
